go/parquet/pqarrow/uuid.go

```diff
@@ -0,0 +1,198 @@
+package pqarrow
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/goccy/go-json"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/google/uuid"
+)
+
+type UUIDBuilder struct {
```

Comment:
maybe we should shift the existing UUID extension type out of the `arrow/internal/testing/types` pkg and put it somewhere in the shared `internal` package that both `arrow` and `parquet` packages can use. that way we don't have to duplicate the code.

Not saying you have to, just an idea. I'm also fine with having this separately in here, though since it is only used for testing I'd prefer to keep this to a `_test` package or a `_test.go` file if you don't shift it to the shared internal package. Basically I don't want this to be externally accessible please.